components/filter/demo/filter-search-demo.js

```diff
@@ -40,6 +40,11 @@ class FilterSearchDemo extends LitElement {
 						<d2l-filter-dimension-set-value key="${value.key}" text="${value.text}" ?selected="${value.selected}"></d2l-filter-dimension-set-value>
 					`)}
 				</d2l-filter-dimension-set>
+				<d2l-filter-dimension-set key="event-initial-subset" text="Event on Search - Initial Subset" search-type="manual" header-text="Related Roles at D2L" selected-first>
+					${this._fullDataSingle.map(value => html`
```

Comment:
Missed this the first time - does this need it's own data array? The demo seems fine, but I'm surprised setting things in this demo isn't affecting the `event-single` one, based on the code in `_handleFilterChange` 🤷

Reply:
Fixed the test but unsure if the demos are setup as we expect them.

The way the demos are setup it will correctly keep track of the data from those filters but changes will not be reflected on the filter. If we want that to be the case, then we would need to change the arrays to state properties and assign new arrays to those properties on filter changes, as opposed to altering the existing array.

That's why we couldn't actually see that bug affecting the demo.

Reply:
That's annoying, I guess I tried to setup the demo as an example of what a consumer needs to do, but only half set it up 😒. Does `this.requestUpdate()` work instead of having to set to new arrays?

Reply:
Yep, so `requestUpdate()` does work. I also needed to deep copy the `initialData` since it was previously using the same objects even after copying the array.

Reply:
Right, kinda messy but I real consumer shouldn't run into that using actual data from the source.  Thanks for making this demo actually work! 😅